client/rest-high-level/src/main/java/org/elasticsearch/client/enrich/PutPolicyRequest.java

```diff
@@ -0,0 +1,174 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.enrich;
+
+import org.elasticsearch.client.Validatable;
+import org.elasticsearch.client.ValidationException;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.index.query.QueryBuilder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class PutPolicyRequest implements Validatable, ToXContentObject {
+
+    static final ParseField TYPE_FIELD = new ParseField("type");
+    static final ParseField QUERY_FIELD = new ParseField("query");
+    static final ParseField INDICES_FIELD = new ParseField("indices");
+    static final ParseField ENRICH_KEY_FIELD = new ParseField("enrich_key");
+    static final ParseField ENRICH_VALUES_FIELD = new ParseField("enrich_values");
+
+    private String name;
+    private String type;
+    private BytesReference query;
+    private List<String> indices;
+    private String enrichKey;
+    private List<String> enrichValues;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public BytesReference getQuery() {
+        return query;
+    }
+
+    public void setQuery(BytesReference query) {
+        this.query = query;
+    }
+
+    public void setQuery(QueryBuilder query) throws IOException {
+        setQuery(xContentToBytes(query));
+    }
+
+    public List<String> getIndices() {
+        return indices;
+    }
+
+    public void setIndices(List<String> indices) {
+        this.indices = indices;
+    }
+
+    public String getEnrichKey() {
+        return enrichKey;
+    }
+
+    public void setEnrichKey(String enrichKey) {
+        this.enrichKey = enrichKey;
+    }
+
+    public List<String> getEnrichValues() {
+        return enrichValues;
+    }
+
+    public void setEnrichValues(List<String> enrichValues) {
+        this.enrichValues = enrichValues;
+    }
+
+    @Override
+    public Optional<ValidationException> validate() {
+        final ValidationException validationException = new ValidationException();
+        if (Strings.hasLength(name) == false) {
```

Comment:
these that are required should be stuffed in a constructor w/ all of the validation in the constructor itself rather than here